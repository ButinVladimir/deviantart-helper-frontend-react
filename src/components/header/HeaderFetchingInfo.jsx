import React from 'react';
import PropTypes from 'prop-types';
import Navbar from 'react-bulma-components/lib/components/navbar';
import Button from 'react-bulma-components/lib/components/button';

export default function HeaderFetchingInfo({
  revokeLoading,
  revokeHandler,
}) {
  return (
    <Navbar.Menu>
      <Navbar.Container position="end">
        <Navbar.Item>
          <Button.Group>
            <Button id="revoke-button" color="danger" loading={revokeLoading} onClick={revokeHandler}>Log out</Button>
          </Button.Group>
        </Navbar.Item>
      </Navbar.Container>
    </Navbar.Menu>
  );
}

HeaderFetchingInfo.propTypes = {
  revokeLoading: PropTypes.bool.isRequired,
  revokeHandler: PropTypes.func.isRequired,
};
