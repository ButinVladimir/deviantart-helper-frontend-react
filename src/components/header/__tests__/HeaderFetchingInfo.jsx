import React from 'react';
import { shallow } from 'enzyme';
import HeaderFetchingInfo from '../HeaderFetchingInfo';

describe('HeaderFetchingInfo', () => {
  it('can be rendered correctly', () => {
    const wrapper = shallow(<HeaderFetchingInfo revokeLoading={false} revokeHandler={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('can handle revoking session', () => {
    const revokeHandler = jest.fn();
    const wrapper = shallow(
      <HeaderFetchingInfo revokeLoading={false} revokeHandler={revokeHandler} />,
    );

    wrapper.find('#revoke-button').simulate('click');

    expect(revokeHandler).toHaveBeenCalled();
  });
});
