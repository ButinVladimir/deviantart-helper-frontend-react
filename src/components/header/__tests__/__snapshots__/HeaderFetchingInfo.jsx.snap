// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFetchingInfo can be rendered correctly 1`] = `
<NavbarMenu
  className=""
  style={Object {}}
>
  <NavbarContainer
    className=""
    position="end"
    renderAs="div"
    style={Object {}}
  >
    <NavbarItem
      active={false}
      className=""
      dropdown={false}
      dropdownUp={false}
      hoverable={false}
      renderAs="a"
      style={Object {}}
    >
      <ButtonGroup
        renderAs="div"
      >
        <Button
          className=""
          color="danger"
          disabled={false}
          fullwidth={false}
          id="revoke-button"
          inverted={false}
          isSelected={false}
          isStatic={false}
          loading={false}
          onClick={[Function]}
          outlined={false}
          remove={false}
          renderAs="button"
          reset={false}
          rounded={false}
          size={null}
          state={null}
          style={Object {}}
          submit={false}
          text={false}
        >
          Log out
        </Button>
      </ButtonGroup>
    </NavbarItem>
  </NavbarContainer>
</NavbarMenu>
`;
